protocol/communities/manager.go

```diff
@@ -1353,10 +1381,18 @@ func (m *Manager) reevaluateMemberChannelsPermissions(community *Community, memb
 	return addToChannels, removeFromChannels, nil
 }
 
-func (m *Manager) checkChannelsPermissions(channelsPermissionsPreParsedData map[string]*PreParsedCommunityPermissionsData, accountsAndChainIDs []*AccountChainIDsCombination, shortcircuit bool) (map[string]map[protobuf.CommunityTokenPermission_Type]bool, error) {
+func (m *Manager) checkChannelsPermissionsImpl(channelsPermissionsPreParsedData map[string]*PreParsedCommunityPermissionsData, accountsAndChainIDs []*AccountChainIDsCombination, shortcircuit bool, collectiblesOwners CollectiblesOwners) (map[string]map[protobuf.CommunityTokenPermission_Type]bool, error) {
+	checkPermissions := func(channelsPermissionPreParsedData *PreParsedCommunityPermissionsData) (*CheckPermissionsResponse, error) {
+		if collectiblesOwners != nil {
+			return m.PermissionChecker.CheckPermissionsWithPreFetchedData(channelsPermissionPreParsedData, accountsAndChainIDs, true, collectiblesOwners)
+		} else {
+			return m.PermissionChecker.CheckPermissions(channelsPermissionPreParsedData, accountsAndChainIDs, true)
```

Comment:
I guess a further improvement would be to do the same for ERC20s? Is there a similar `getOwners` function for ERC20?

Reply:
`getOwners` for ERC20 has much higher risk to be space inefficient. Imagine getting all owners of SNT token :thinking:

Reply:
Yeah that's what I thought.